query/regexp.go

```diff
@@ -74,15 +77,19 @@ func regexpToQueryRecursive(r *syntax.Regexp, minTextSize int) Q {
 
 	case syntax.OpConcat, syntax.OpAlternate:
 		var qs []Q
+		isSym := true
 		for _, sr := range r.Sub {
-			if sq := regexpToQueryRecursive(sr, minTextSize); sq != nil {
+			if sq, sm := regexpToQueryRecursive(sr, minTextSize); sq != nil {
+				if !sm {
+					isSym = false
+				}
 				qs = append(qs, sq)
 			}
 		}
 		if r.Op == syntax.OpConcat {
-			return &And{qs}
+			return &And{qs}, isSym
```

Comment:
I think OpConcat should probably return false if there is more than 1 field. What is an example of `OpConcat` which would return the same results? eg say we have `(foo)(bar)`. `foo AND bar` returns different results to `foobar`.

Reply:
It makes sense 👍